Content.Server/GameObjects/Components/ActionBlocking/CuffedComponent.cs

```diff
@@ -0,0 +1,351 @@
+                !EntitySystem.Get<SharedInteractionSystem>().InRangeUnobstructed(
+                    user.Transform.MapPosition,
+                    Owner.Transform.MapPosition,
+                    _interactRange,
+                    ignoredEnt: Owner))
+            {
+                user.PopupMessage(user, "You are too far away to remove the cuffs.");
+                return;
+            }
+
+            if (!EntitySystem.Get<SharedInteractionSystem>().InRangeUnobstructed(
+                    cuffsToRemove.Transform.MapPosition,
+                    Owner.Transform.MapPosition,
+                    _interactRange,
+                    ignoredEnt: Owner))
+            {
+                Logger.Warning($"Handcuffs being removed from player are obstructed or too far away! This should not happen!");
+                return;
+            }
+
+            user.PopupMessage(user, "You start removing the cuffs.");
+
+            var audio = EntitySystem.Get<AudioSystem>();
+            audio.PlayFromEntity(isOwner ? cuff.StartBreakoutSound : cuff.StartUncuffSound, Owner);
+
+            var uncuffTime = isOwner ? cuff.BreakoutTime : cuff.UncuffTime;
+            var doAfterEventArgs = new DoAfterEventArgs(user, uncuffTime)
+            {
+                BreakOnUserMove = true,
+                BreakOnDamage = true,
+                BreakOnStun = true,
+                NeedHand = true
+            };
+
+            var doAfterSystem = EntitySystem.Get<DoAfterSystem>();
+            var result = await doAfterSystem.DoAfter(doAfterEventArgs);
+
+            if (result != DoAfterStatus.Cancelled)
+            {
+                audio.PlayFromEntity(cuff.EndUncuffSound, Owner);
+
+                _container.ForceRemove(cuffsToRemove);
+                cuffsToRemove.Transform.AttachToGridOrMap();
+                cuffsToRemove.Transform.WorldPosition = Owner.Transform.WorldPosition;
+
+                if (cuff.BreakOnRemove)
+                {
+                    cuff.Broken = true;
+
+                    cuffsToRemove.Name = cuff.BrokenName;
+                    cuffsToRemove.Description = cuff.BrokenDesc;
+
+                    if (cuffsToRemove.TryGetComponent<SpriteComponent>(out var sprite))
+                    {
+                        sprite.LayerSetState(0, cuff.BrokenState); // TODO: safety check to see if RSI contains the state?
+                    }
+                }
+
+                CanStillInteract = _hands.Hands.Count() > CuffedHandCount;
+                OnCuffedStateChanged.Invoke();
+                UpdateStatusEffect();
+                Dirty();
+
+                if (CuffedHandCount == 0)
+                {
+                    _notifyManager.PopupMessage(user, user, Loc.GetString("You successfully remove the cuffs."));
+
+                    if (!isOwner)
+                    {
+                        _notifyManager.PopupMessage(user, Owner, Loc.GetString($"{0} uncuffs your hands.", user.Name));
```

Comment:
Same here

Reply:
0:theName doesn't seem to work on players as intended